app/controllers/audits_controller.rb

```diff
@@ -0,0 +1,37 @@
+class AuditsController < ApplicationController
+  def show
+    audit.questions = audit.template.questions if audit.new_record?
+  end
+
+  def save
+    audit.user = current_user
+
+    if audit.update(audit_params)
+      flash.now.notice = "Audit saved successfully."
```

Comment:
I tried this initially, but we lose the user's pre-filled form data if we redirect. I don't think there's a way to preserve that when redirecting?